### PR TITLE
rtty: Add package

### PR DESCRIPTION
**Pull request by zhaojh329**
Maintainer: me
Compile tested: (mipsel,miwifi-mini, LEDE 60a39e8f5a)
Run tested: (mipsel,miwifi-mini, LEDE 60a39e8f5a)

Description:
https://github.com/zhaojh329/rtty
Access your terminal over the web browser. The 'r' in the name refers to the 'Reverse Proxy' or 'Remote'. It is composed of the client and the server.

### CODE REVIEW

**Comment by zhaojh329** [comment]
How long do I have to wait for it to be merged or get a reply?

File: libs/libuwsc/Makefile [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (C) 2014-2017 OpenWrt.org
```
Comment (yousong) [inline libs/libuwsc/Makefile]: Though I am not a lawyer, it's very likely that you should use year 2018 and your own name in the copyright header.

File: libs/libuwsc/Makefile [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+PKG_LICENSE:=GPL-3.0
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_MAINTAINER:=Jianhui Zhao <jianhuizhao329@gmail.com>
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+CMAKE_OPTIONS += -DUWSC_DEBUG=on
```
Comment (yousong) [inline libs/libuwsc/Makefile]: Debug option should be off by default.

File: libs/libuwsc/Makefile [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+define Package/libuwsc/default
+  SECTION:=libs
+  CATEGORY:=Libraries
+  SUBMENU:=Networking
+  #TITLE:=Lightweight WebSocket client library based on libubox
+  TITLE:=Lightweight WebSocket client library
+  DEPENDS:=+libubox
+endef
```
Comment (yousong) [inline libs/libuwsc/Makefile]: Such commented-out code should be removed

File: utils/rtty/Makefile [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (C) 2014-2017 OpenWrt.org
```
Comment (yousong) [inline utils/rtty/Makefile]: ditto

File: utils/rtty/Makefile [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/rtty
+  SECTION:=utils
+  CATEGORY:=Utilities
+  SUBMENU:=Terminal
+  TITLE:=Access your terminal over the web browser
+  DEPENDS:=+libuwsc-nossl +libblobmsg-json
```
Comment (yousong) [inline utils/rtty/Makefile]: I am not quite sure, but if rtty can work with all those variants of libuwsc, please consider testing whether letting those variants `PROVIDES:=libuwsc` then `DEPENDS:=+libuwsc`.  Like the relationship between `iproute2` and `wireguard`.

File: utils/rtty/Makefile [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+endef
+
+define Package/rtty/description
+Access your terminal over the web browser. The 'r' in the name refers to the 'Reverse Proxy' or 'Remote'.
+It is composed of the client and the server. You can access any of your terminals through a web browser based on the MAC address.
+endef
```
Comment (yousong) [inline utils/rtty/Makefile]: Paragraphs should be separated by an empty line.  Better if they are line wrapped at 80 chars length ;)

File: utils/rtty/files/rtty.config [hunk 7]
```diff
@@ -0,0 +1,6 @@
+config device
+    option interface 'lan'
+
+config server
+  	option	host 'jianhuizhao.f3322.net'
+  	option	port '5912'
```
Comment (yousong) [inline utils/rtty/files/rtty.config]: Maybe the `interface` option can also reside in the `server` section?

Comment (yousong) [reply utils/rtty/files/rtty.config]: Sample config should not refer to any specific server and should be off by default.

**Comment by zhaojh329** [comment]
I'll close the PR and Resubmit.

**Comment by yousong** [comment]
@zhaojh329 , in case it may help, it's possible to update the current pull request by force-push to your source branch.  It would be better if things of the same topic are in a single place ;)

**Comment by zhaojh329** [comment]
ok,thanks.

**Comment by zhaojh329** [comment]
I'm sorry,I already resubmit.

**Pull request closed without merging**